xla/service/gpu/BUILD

```diff
@@ -919,13 +919,13 @@ alias(
 
 cc_library(
     name = "_nccl_api_impl",
-    srcs = if_cuda_is_configured(
+    srcs = if_cuda_or_rocm(
```

Comment:
This changes the requirement that we need to compile with --config=cuda or --config=rocm. I think we want to avoid that. Unfortunately if_gpu_is_configured has no else branch. Recently, another contributor worked around this by putting the code from the stub file into the other file:

https://github.com/openxla/xla/pull/8285/files

Maybe that is also the easiest thing to do here.

Reply:
Hmm... How do I see the CI error? The public CI completes.

Reply:
I think @akuegel referring to their Google internal CI. Basically all XLA PRs will have another internal CI checks before it's merged.

Reply:
Hi @akuegel just to be sure IIUC, to resolve this compile config issue, you expect to move the whole nccl_api_stub.cc into nccl_api.cc like this part ? https://github.com/openxla/xla/pull/8285/files#diff-e5b27680b58478fd5e3ac3bf7d3d5aba0c376aece9566f015a9beaa7f46c60dfL121-R130

Reply:
Yes, unless you have a better idea :)
In any case, using if_cuda_or_rocm will not work, and if_gpu_is_configured has no support for an "else" branch.

Reply:
Thanks for clarification! I will do it in another PR with some bug fixes on this.

Reply:
I notice this is PASSED in internal checks, I assume this PR will be merged first? Then, I will make change in another PR to resolve this config issue. Or I create another PR and cherry-pick this one with the changes (so this one can be closed?)

Reply:
After I commented on the PR, the internal version of this PR got recreated, I don't know why. Normally this should only happen with an approval of the PR. We are not planning to merge the PR as-is.

Reply:
I c. Thanks for clarification! I will do another freash one with the changes.

Reply:
Thank you